src/mod/applications/mod_http_cache/prometheus_metrics.cpp

```diff
@@ -0,0 +1,157 @@
+#include <map>
+#include <string>
+#include "prometheus_metrics.h"
+
+
+class prometheus_metrics
+{
+public:
+	class auto_lock
+	{
+	public:
+		auto_lock(switch_mutex_t* mutex) : _mutex(mutex)
+		{
+			switch_mutex_lock(_mutex);
+		}
+		~auto_lock()
+		{
+			switch_mutex_unlock(_mutex);
+		}
+		switch_mutex_t* _mutex;
+	};
+	
+	prometheus_metrics(switch_loadable_module_interface_t **module_interface, switch_memory_pool_t* pool) :
+		_pool(pool),
+		_module_interface(module_interface),
+		_download_fail_count(0),
+		_download_bucket_0_300(0),
+		_download_bucket_300_500(0),
+		_download_bucket_500_800(0),
+		_download_bucket_800_1300(0),
+		_download_bucket_1300_2100(0),
+		_download_bucket_2100_3400(0),
+		_download_bucket_3400_5500(0),
+		_download_bucket_5500_inf(0),
+		_download_bucket_sum(0),
+		_download_bucket_count(0)
+	{
+		switch_mutex_init(&_mutex, SWITCH_MUTEX_NESTED, _pool);
+	}
+	
+	~prometheus_metrics()
+	{
+		switch_mutex_destroy(_mutex);
+	}
+
+	void increment_download_duration(unsigned int duration)
+	{
+		auto_lock lock(_mutex);
+
+		_download_bucket_sum += duration;
+		_download_bucket_count++;
+
+		if (duration < 300) {
+			_download_bucket_0_300++;
+		} else if (300 <= duration && duration < 500) {
```

Comment:
First the left condition should be greater than and not less than. Next, you don't need the left condition here. I think its enough to just do `duration < 500`. Why? Because it should have been catch on the previous if statement.

Reply:
Agree, I'll fix it. :)

Reply:
Fixed it. :)